test/LdapTest.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Copyright (c)2015-2015 heiglandreas
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIBILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @category
+ * @author    Andreas Heigl<andreas@heigl.org>
+ * @copyright ©2015-2015 Andreas Heigl
+ * @license   http://www.opesource.org/licenses/mit-license.php MIT-License
+ * @version   0.0
+ * @since     10.07.15
+ * @link      https://github.com/heiglandreas/
+ */
+
+namespace ZendTest\Ldap;
+
+use phpmock\phpunit\PHPMock;
+use Zend\Ldap\Dn;
+use Zend\Ldap\Exception\LdapException;
+use Zend\Ldap\Ldap;
+
+class LdapTest extends \PHPUnit_Framework_TestCase
+{
+    use PHPMock;
+
+    public function setup()
+    {
+        if (!getenv('TESTS_ZEND_LDAP_ONLINE_ENABLED')) {
+            $this->markTestSkipped("Zend_Ldap online tests are not enabled");
+        }
+    }
+
+    /**
+     * @param $dn
+     * @param $attribs
+     * @param $sure
+     *
+     * @dataProvider removingAttributesProvider
+     * @throws \Zend\Ldap\Exception\LdapException
+     */
+    public function testRemovingAttributes($dn, $attribs, $sure, $expectedDn, $expectedArray)
+    {
+        $ldap_mod_del = $this->getFunctionMock('Zend\\Ldap', "ldap_mod_del");
```

Comment:
Sorry, but PHPUnit only mocks methods, not functions :wink: I could write this test without the functionMock but as there might be more tests mocking the ldap_functions it might be a good idea to include the framework